### PR TITLE
[pkg/otlp] Default OTLP logs ingestion to disabled

### PR DESCRIPTION
**Pull request by IbraheemA**
### What does this PR do?

Previously, OTLP log ingestion was enabled by default. So, when logs-agent was not enabled as well, the agent would crash (logs-agent needs to be running to use logs ingestion.)

Previously, when `otlp_config: logs: ` was included in the config, logs would be enabled; now `otlp_config: logs: enabled: true` must be specified in full.

### Describe how to test/QA your changes

Build agent locally, and run it with various settings of `logs_enabled` to verify the following:

1. When logs is not present in otlp_config, does the OTel pipeline start successfully
2. The env var can still enable otlp log ingestion
3. If env is not set, all of `otlp_config: logs: enabled: true` must be included in the yaml to enable logs

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/default_otlp_logs_enabled_false-6d98a23f7a76f153.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    OTLP logs ingestion is now disabled by default, and otlp_config: logs: enabled: true must be specified to enable
```
Comment (brett0000FF) [inline releasenotes/notes/default_otlp_logs_enabled_false-6d98a23f7a76f153.yaml:11]:
```suggestion
    OTLP logs ingestion is now disabled by default. To enable it, set otlp_config.logs.enabled to true.
```
I think dot notation is a bit easier to read ^

**Review by brett0000FF — CHANGES_REQUESTED** [review]
Small docs suggestion for clarity in the release note.

**Review by songy23 — COMMENTED** [review]
Have you verified:
1. When logs is not present in otlp_config, does the OTel pipeline start successfully
2. The env var can still enable otlp log ingestion

**Comment by IbraheemA** [comment]
> Have you verified:
> 
> 1. When logs is not present in otlp_config, does the OTel pipeline start successfully
> 2. The env var can still enable otlp log ingestion

Yes, added these to the QA section

**Review by brett0000FF — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]

File: pkg/otlp/pipeline_validator_serverless.go [hunk 2]
```diff
@@ -10,15 +10,13 @@ package otlp
...
 	"fmt"
 
-	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/logs/message"
 )
 
-func checkAndUpdateCfg(cfg config.Config, pcfg *PipelineConfig, logsAgentChannel chan *message.Message) error {
-	if HasLogsSectionEnabled(cfg) {
+func checkAndUpdateCfg(pcfg PipelineConfig, logsAgentChannel chan *message.Message) error {
+	if pcfg.LogsEnabled(cfg) {
```
Comment (songy23) [inline pkg/otlp/pipeline_validator_serverless.go:17]:
```suggestion
	if pcfg.LogsEnabled {
```

**Review by IvanTopolcic — APPROVED** [review]

**Pull request merged**